src/linear_algebra/affine_space.lean

```diff
@@ -358,30 +353,91 @@ instance : inhabited (affine_subspace k V P) := ⟨univ k V P⟩
 
 variables {k V P}
 
+/-- The direction of an affine subspace is the submodule spanned by
+the pairwise differences of points.  (Except in the case of an empty
+affine subspace, where the direction is the zero submodule, every
+vector in the direction is the difference of two points in the affine
+subspace.) -/
+def direction (s : affine_subspace k V P) : submodule k V := vector_span k V (s : set P)
+
+/-- The direction equals the `vector_span`. -/
+lemma direction_eq_vector_span (s : affine_subspace k V P) :
+  s.direction = vector_span k V (s : set P) :=
+rfl
+
+/-- Alternative definition of the direction when the affine subspace
+is nonempty.  This is only intended to be used in the proof of
+`direction_coe_eq_vsub_set`. -/
+def direction_of_nonempty {s : affine_subspace k V P} (h : (s : set P).nonempty) :
```

Comment:
What is the benefit of this (longish) definition, if you afterwards prove that it is exactly the same submodule?
The carriers are not defeq, but would it be sufficient to just have a lemma giving the alternative carrier under the `nonempty` assumption?

Oooh, that's actually what `direction_coe_eq_vsub_set` is all about...

Reply:
Indeed, the point of this definition is so that the order on submodules can be used (given that `submodule.span` is defined in terms of that order) in proving (via `le_antisymm`) what are now `coe_direction_eq_vsub_set` and thus `mem_direction_iff_eq_vsub` and other lemmas.

Reply:
Yes, I get it now. Maybe you could add that explanation explicitly to the docstring (or a regular comment).

Reply:
Expanded the docstring.